addons/udes_stock/models/abstract_stock_model.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+from odoo.osv import expression
+
+
+class AbstractStockModel(models.AbstractModel):
+    _name = 'abstract.stock.model'
+    _description = 'Inherited Model'
+
+    # Whether the model can create or not
+    ASM_CREATE = False
+    # The default domain to search with a str
+    ASM_STR_DOMAIN = ['name']
```

Comment:
Is it safe to use a mutable class variable here?  What would happen if it were modified by an instance or a subclass?

nit: the comment implies a single domain, but presumably there can be more than one.

Reply:
It is designed to be overridden by the class that inherits - as to how safe, suppose it is until it isn't. The main idea was to not have to pass lots of variables, and do extra checks as a result. But happy to change back, or do something else,  if you feel very strong about it :)